src/hb-ot-var-fvar-table.hh

```diff
@@ -95,20 +92,57 @@ struct AxisRecord
     info->tag = axisTag;
     info->name_id = axisNameID;
     info->flags = (hb_ot_var_axis_flags_t) (unsigned int) flags;
-    info->default_value = defaultValue / 65536.f;
-    /* Ensure order, to simplify client math. */
-    info->min_value = hb_min (info->default_value, minValue / 65536.f);
-    info->max_value = hb_max (info->default_value, maxValue / 65536.f);
+    fill_values (info->default_value, info->min_value, info->max_value);
     info->reserved = 0;
   }
 
+  int normalize_axis_value (float v) const
+  {
+    float default_value, min_value, max_value;
+    fill_values (default_value, min_value, max_value);
+
+    v = hb_clamp (v, min_value, max_value);
+
+    if (v == default_value)
+      return 0;
+    else if (v < default_value)
+      v = (v - default_value) / (default_value - min_value);
+    else
+      v = (v - default_value) / (max_value - default_value);
+    return roundf (v * 16384.f);
+  }
+
+  float unnormalize_axis_value (int v) const
+  {
+    float default_value, min_value, max_value;
+    fill_values (default_value, min_value, max_value);
+
+    if (v == 0)
+      return default_value;
+    else if (v < 0)
+      return v * (default_value - min_value) / 16384.f + default_value;
+    else
+      return v * (max_value - default_value) / 16384.f + default_value;
+  }
+
+  hb_ot_name_id_t get_name_id () const { return axisNameID; }
+
   bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
     return_trace (c->check_struct (this));
   }
 
-  public:
+  protected:
+  void fill_values (float &default_value, float &min_value, float &max_value) const
```

Comment:
`fill_values` is a very bad and wrong name. And since you are passing references in, reading the call site it's not even clear which direction data flows.

`get_triplet` or whatever.

Reply:
`get_shape` maybe even. I also like them ordered as `min, default, max`. Same way that for other things we use `min, peak, max`.

Reply:
Done

Reply:
How about `get_coordinates`, that reflect what spec calls them